tests/publiccloud/check_registercloudguest.pm

```diff
@@ -39,80 +39,75 @@ sub run {
+        if ($instance->ssh_script_output(cmd => 'sudo zypper lr') !~ /No repositories defined/gm) {
             die 'The BYOS instance should be unregistered and report "Warning: No repositories defined.".';
         }
 
-        if ($instance->run_ssh_command(cmd => 'sudo systemctl is-enabled guestregister.service', proceed_on_failure => 1) !~ /disabled/) {
-            die('guestregister.service is not disabled');
-        }
+        $instance->ssh_assert_script_run(cmd => '! sudo systemctl is-enabled guestregister.service', fail_message => 'guestregister.service is not disabled');
 
-        if ($instance->run_ssh_command(cmd => 'sudo ls /etc/zypp/credentials.d/ | wc -l', proceed_on_failure => 1) != 0) {
-            $instance->run_ssh_command(cmd => 'sudo ls -la /etc/zypp/credentials.d/', proceed_on_failure => 1);
+        if ($instance->ssh_script_output(cmd => 'sudo ls /etc/zypp/credentials.d/ | wc -l') != 0) {
+            $instance->ssh_script_run(cmd => 'sudo ls -la /etc/zypp/credentials.d/');
             die('/etc/zypp/credentials.d/ is not empty');
         }
 
-        if (is_azure() && $instance->run_ssh_command(cmd => 'sudo systemctl is-enabled regionsrv-enabler-azure.timer', proceed_on_failure => 1) !~ /enabled/) {
+        if (is_azure() && $instance->ssh_sssert_cript_run(cmd => 'sudo systemctl is-enabled regionsrv-enabler-azure.timer')) {
             die('regionsrv-enabler-azure.timer is not enabled');
         }
 
-        if ($instance->run_ssh_command(cmd => 'sudo stat --printf="%s" /var/log/cloudregister', proceed_on_failure => 1) != 0) {
+        if ($instance->ssh_script_output(cmd => 'sudo stat --printf="%s" /var/log/cloudregister') != 0) {
             die('/var/log/cloudregister is not empty');
         }
-        # The `sudo SUSEConnect -d` is not supported on BYOS and should fail.
-        $instance->run_ssh_command(cmd => '! sudo SUSEConnect -d');
+        $instance->ssh_assert_script_run(cmd => '! sudo SUSEConnect -d', fail_message => 'SUSEConnect succeeds but it is not supported should fail on BYOS');
     } else {
-        if ($instance->run_ssh_command(cmd => 'sudo zypper lr | wc -l', proceed_on_failure => 1, timeout => 360) < 5) {
-            record_info('zypper lr', $instance->run_ssh_command(cmd => 'sudo zypper lr', proceed_on_failure => 1));
+        if ($instance->ssh_script_output(cmd => 'sudo zypper lr | wc -l', timeout => 360) < 5) {
+            record_info('zypper lr', $instance->ssh_script_output(cmd => 'sudo zypper lr'));
             die 'The list of zypper repositories is too short.';
         }
 
-        if ($instance->run_ssh_command(cmd => 'sudo systemctl is-enabled guestregister.service', proceed_on_failure => 1) !~ /enabled/) {
-            die('guestregister.service is not enabled');
-        }
+        $instance->ssh_assert_script_run(cmd => 'sudo systemctl is-enabled guestregister.service', fail_message => 'guestregister.service is not enabled');
```

Comment:
we dramatically changing test coverage and test behavior here , let's not do this

Reply:
We're not changing test coverage but rather getting rid of workarounds caused by `run_ssh_command`.

Reply:
`$instance->run_ssh_command(cmd => 'sudo systemctl is-enabled guestregister.service', proceed_on_failure => 1) !~ /enabled/`

test will die ONLY when `systemctl is-enabled guestregister.service`  will output string containing "enabled"

`$instance->ssh_assert_script_run(cmd => 'sudo systemctl is-enabled guestregister.service', fail_message => 'guestregister.service is not enabled');`

ANY none-zero exit code will cause test failure

P.S. I would like to avoid discussion around "how big chances that something else will happen here "

Reply:
Any other non-zero exit code would mean, that we can't say if the service is disabled and thus the state is undefined. In such cases we should fail the test run.

If you can't test a requirement, then the only safe assumption is that it's broken.

Reply:
The `systemctl is-enabled` will return non zero code only when the service is not enabled. I use this check on many tests for several years and I did not record exception so far. I also initiated this test module and the check for `"enabled"` is there as a workaround for my problems with `run_ssh_command`.

Reply:
please look for "is-enabled" in systemctl documentation  https://www.freedesktop.org/software/systemd/man/systemctl.html

additionally to enabled/disabled service can be : 
-  masked
- static
- indirect
- transient
- bad

some of them will causing 0 exit code others none zero this is what I am calling "change of test coverage" .

and I would like to ask again - **let's not speak about "what probability of getting status transient" ? I never saw it in my whole life **  we not betting here . If program has some status we should expect that it will return it

Reply:
so for example "static" return 0 too  BUT static is - " The unit file **is not enabled**, and has no provisions for enabling in the [Install] unit file section."

Reply:
> I use this check on many tests for several years and I did not record exception so far.

who knows maybe you have missed some service in "static" state due to taking such approach

Reply:
I really think this one is fine as it is. If it works, everything is fine, if it fails for whatever reason the test fails. What more do we want?

Reply:
so you don't care if service is enabled or it is "static" state with broken service file ?